KAFKA-8190; Don't update keystore modification time during validation

rajinisivaram opened the pull request:
We currently store keystore file modification time when loading keystores in a `SecurityStore` instance. When dynamically updating keystores without filename change, we compare the time at the last load against the current file modification time. But we load keystore for validation of dynamic configs and as a result, we dont recreate SSLContext when performing actual reconfiguration after the validation. We always create a new `SecurityStore` instance for reconfiguration of the store, so we only need to store file modification time when we construct the instance.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

rajinisivaram committed:
KAFKA-8190; Don't update keystore modification time during validation

omkreddy reviewed (approved):
@rajinisivaram Thanks for the PR. Good find. LGTM.
looks like we don't have a test in DynamicBrokerReconfigurationTest to test this scenario.

rajinisivaram committed:
Update DynamicBrokerReconfigurationTest

rajinisivaram commented:
@omkreddy Thanks for the review. Updated DynamicBrokerReconfigurationTest to test this scenario. Can you take a look? Thanks.

omkreddy commented:
Thanks for the update. LGTM.

rajinisivaram committed:
Update file modification time to avoid transient failures, log at info level

rajinisivaram commented:
There was a test failure in one of the builds. Looks like this was because file modification time hadn't changed on the file. Updated test to change file modification time explicitly (I could recreate the failure by running in a loop and it passes reliably for me).

Also added info-level log entry for SSLContext update. For all config changes, we log at info-level in `DynamicBrokerConfig`. But since keystore update can happen without config change, it will be useful to have the reload logged at info level as well.

@omkreddy Can you review once more please? Sorry about that.

omkreddy commented:
Thanks for the update. LGTM.

rajinisivaram commented:
@omkreddy Thanks for the reviews! Merging to trunk, 2.2 and 2.1. Build failure is unrelated (ConsumerBounceTest).
